models/user.go

```diff
@@ -394,20 +392,22 @@ func (u *User) RealSizedAvatarLink(size int) string {
 		return base.DefaultAvatarLink()
 	}
 
+	fs := storage.Avatar(context.Background(), u.Avatar)
+
 	switch {
 	case u.UseCustomAvatar:
-		if !com.IsFile(u.CustomAvatarPath()) {
+		if !fs.Exists() {
 			return base.DefaultAvatarLink()
 		}
-		return setting.AppSubURL + "/avatars/" + u.Avatar
+		return setting.AppSubURL + "/avatars?obj=" + u.Avatar
```

Comment:
Is the router change necessary?

Reply:
Hmm, I'm not sure about this change, I thought it dues to the rebase to current master

Reply:
I re-check, it comes from original commit by @masudur-rahman, look at the Attachment repository, seems we don't use it anyway

Reply:
Oh, we did, as avatars and repo-avatars are handle in them same handler:

```
func (opts *AvatarOptions) handle(ctx *macaron.Context, log *log.Logger) bool {
	if ctx.Req.Method != "GET" && ctx.Req.Method != "HEAD" {
		return false
	}

	avatarURL := ctx.Req.URL
	if opts.Prefix != avatarURL.Path {
		return false
	}
	objPath := avatarURL.Query().Get("obj")
```